modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/AdminTooltip.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayButton from '@clayui/button';
+import ClayCard from '@clayui/card';
+import {useResource} from '@clayui/data-provider';
+import ClayDropDown from '@clayui/drop-down';
+import ClayForm, {ClayInput, ClayRadio, ClayRadioGroup} from '@clayui/form';
+import PropTypes from 'prop-types';
+import React, {useLayoutEffect, useState} from 'react';
+
+const AdminTooltip = ({
+	namespace,
+	setRemovePinHandler,
+	setShowTooltip,
+	showTooltip,
+	updatePin,
+}) => {
+	const [pinLabel, setPinLabel] = useState(showTooltip.details.label);
+	const [linkedValue, setLinkedValue] = useState(
+		showTooltip.details.linked_to_sku
+	);
+	const [sku, setSku] = useState(showTooltip.details.sku);
+	const [quantity, setQuantity] = useState(showTooltip.details.quantity);
+	const [networkStatus, setNetworkStatus] = useState(4);
+	const initialLoading = networkStatus === 1;
+	const loadingAd = networkStatus < 4;
+	const error = networkStatus === 5;
+
+	const {resource} = useResource({
+		fetchPolicy: 'cache-first',
+		link: 'https://rickandmortyapi.com/api/character/',
+		onNetworkStatusChange: setNetworkStatus,
+		variables: {
+			name: sku,
+		},
+	});
```

Comment:
This endpoint should not be here

[See previous comment](https://github.com/liferay-frontend/liferay-portal/pull/1318/files#r685012590)

Reply:
Thank you @bryceosterhaus. That was only an example pasted from clay doc to test the behaviour. I'm trying to make calls to the real api right now but i can't find any reference about adding the authentication there the make the data-provider and autocomplete to retrieve the datas. Can you suggest how to move forward? ty

Reply:
No need if you use Liferay.fetch that is basically the only solution. If you want to use fetch or other http libraries... you would had to provide the CSRF token and the session id but we should avoid it.

Reply:
Hello i did all this changes in the [next pr](https://github.com/liferay-frontend/liferay-portal/pull/1374) that will fix, using fetch the right api call. So please let it go on. ty